pep-0755.rst

```diff
@@ -0,0 +1,551 @@
+PEP: 755
+Title: Documentation Metadata in Typing
+Author: Sebastián Ramírez <tiangolo@gmail.com>
+Sponsor: Jelle Zijlstra <jelle.zijlstra@gmail.com>
+Discussions-To:
+Status: Draft
+Type: Standards Track
+Topic: Typing
+Content-Type: text/x-rst
+Created: 28-Aug-2023
+Python-Version: 3.13
+Post-History:
+
+
+Abstract
+========
+
+This document proposes a way to complement docstrings to add additional documentation
+to Python symbols using type annotations with ``Annotated`` (in class attributes,
+function and method parameters, return values, and variables).
+
+
+Motivation
+==========
+
+The current standard method of documenting code APIs in Python is using docstrings.
+But there's no standard way to document parameters in docstrings.
+
+There are several pseudo-standards for the format in these docstrings, and new
+pseudo-standards can appear easily: numpy, Google, Keras, reST, etc.
+
+All these formats are some specific syntax inside a string. Because of this, when
+editing those docstrings, editors can't easily provide support for autocompletion,
+inline errors for broken syntax, etc.
+
+Editors don't have a way to support all the possible micro languages in the docstrings
+and show nice user interfaces when developers use libraries with those different
+formats. They could even add support for some of the syntaxes, but probably not all,
+or not completely.
+
+Because the docstring is in a different place in the code than the actual parameters
+and it requires duplication of information (the parameter name) the information about
+a parameter is easily in a place in the code quite far away from the declaration of
+the actual parameter. This means it's easy to refactor a function, remove a parameter,
+and forget to remove its docs. The same happens when adding a new parameter, it's easy
+to forget to add the docstring for it.
+
+Editors can't check or ensure the consistency of the parameters and their docs.
+
+It's not possible to programatically test these docstrings, for example to ensure
+documentation consistency for the parameters across several similar functions, at
+least not without implementing a custom syntax parser.
+
+Some of these formats tried to account for the lack of type annotations in older
+Python versions, but now that information doesn't need to be there.
```

Comment:
I don't understand this sentence, maybe you mean something like "compatibility with Python versions without annotations is no longer important"?

Reply:
Ah! I was referring to the docstring formats that included type information in the docstring. I just rewrote that a bit to try and make it clearer.

Reply:
Thanks, your rewording makes that clearer.